src/mlpack/tests/bayesian_linear_regression_test.cpp

```diff
@@ -0,0 +1,156 @@
+    BOOST_REQUIRE_CLOSE(predictions[i], y[i], 1e-6);
+
+  // Check that the estimated variance is zero.
+  BOOST_REQUIRE_SMALL(estimator.Variance(), 1e-6);
+}
+
+// Verify fitIntercept and normalize equal false do not affect the solution.
+BOOST_AUTO_TEST_CASE(TestCenter0Normalize0)
+{
+  arma::mat X;
+  arma::rowvec y;
+  size_t nDims = 30, nPoints = 100;
+
+  GenerateProblem(X, y, nPoints, nDims, 0.5);
+
+  BayesianLinearRegression estimator(false, false);
+
+  estimator.Train(X, y);
+
+  // To be neutral dataOffset must be all 0.
+  BOOST_REQUIRE(sum(estimator.DataOffset()) == 0.0);
+
+  // To be neutral responseOffset must be 0.
+  BOOST_REQUIRE(estimator.ResponsesOffset() == 0);
+
+  // To be neutral dataScale must be all 1.
+  BOOST_REQUIRE(sum(estimator.DataScale()) == nDims);
+}
+
+// Verify that centering and normalization are correct.
+BOOST_AUTO_TEST_CASE(TestCenter1Normalize1)
+{
+  arma::mat X;
+  arma::rowvec y;
+  size_t nDims = 30, nPoints = 100;
+  GenerateProblem(X, y, nPoints, nDims, 0.5);
+
+  BayesianLinearRegression estimator(true, true);
+  estimator.Train(X, y);
+
+  arma::colvec xMean = arma::mean(X, 1);
+  arma::colvec xStd = arma::stddev(X, 0, 1);
+  double yMean = arma::mean(y);
+
+  BOOST_REQUIRE_SMALL((double) abs(sum(estimator.DataOffset() - xMean)), 1e-6);
+  BOOST_REQUIRE_SMALL((double) abs(estimator.ResponsesOffset() - yMean), 1e-6);
+  BOOST_REQUIRE_SMALL((double) abs(sum(estimator.DataScale() - xStd)), 1e-6);
+}
+
+// Check that Train() does not fail with two colinear vectors.
+BOOST_AUTO_TEST_CASE(SingularMatix)
+{
+  arma::mat X;
+  arma::rowvec y;
+
+  GenerateProblem(X, y, 200, 10);
+  // Now the first and the second rows are indentical.
+  X.row(1) = X.row(0);
+
+  BayesianLinearRegression estimator;
+  double singular = estimator.Train(X, y);
+  BOOST_REQUIRE(singular != -1);
+}
+
+// Check that std are well computed/coherent. At least higher than the
+// estimated predictive variance.
+BOOST_AUTO_TEST_CASE(PredictiveUncertainties)
+{
+  arma::mat X;
+  arma::rowvec y;
+
+  GenerateProblem(X, y, 100, 10, 1);
+
+  BayesianLinearRegression estimator(true, true);
+  estimator.Train(X, y);
+
+  arma::rowvec responses, std;
+  estimator.Predict(X, responses, std);
+  const double estStd = sqrt(estimator.Variance());
+
+  for (size_t i = 0; i < X.n_cols; i++)
+    BOOST_REQUIRE(std[i] > estStd);
+}
+
+// Check the solution is equal to the classical ridge.
+BOOST_AUTO_TEST_CASE(EqualtoRidge)
+{
+  arma::mat X;
+  arma::rowvec y;
+
+  GenerateProblem(X, y, 100, 10, 1);
+
+  BayesianLinearRegression bayesLinReg(false, false);
+  bayesLinReg.Train(X, y);
+
+  LinearRegression classicalRidge(X,
+                                  y,
+                                  bayesLinReg.Alpha() / bayesLinReg.Beta(),
+                                  false);
+  double equalSol = arma::sum(bayesLinReg.Omega()
+                              - classicalRidge.Parameters());
+  BOOST_REQUIRE(equalSol < 1e-5);
```

Comment:
Is it worth testing that the predictions are similar too?  I don't see `Predict()` tested in this test suite, but it's probably good to make sure we test it somehow.